x-pack/plugins/cases/server/client/cases/get.ts

```diff
@@ -232,6 +234,96 @@ export const get = async (
   }
 };
 
+/**
+ * Retrieves a case resolving its ID and optionally loading its comments and sub case comments.
+ *
+ * @experimental
+ */
+export const resolve = async (
+  { id, includeComments, includeSubCaseComments }: GetParams,
+  clientArgs: CasesClientArgs
+): Promise<CaseResolveResponse> => {
+  const { unsecuredSavedObjectsClient, caseService, logger, authorization } = clientArgs;
+
+  try {
+    if (!ENABLE_CASE_CONNECTOR && includeSubCaseComments) {
+      throw Boom.badRequest(
+        'The `includeSubCaseComments` is not supported when the case connector feature is disabled'
+      );
+    }
+
+    const theResolvedCase: SavedObjectsResolveResponse<CaseAttributes> = await caseService.getResolveCase(
+      {
+        unsecuredSavedObjectsClient,
+        id,
+      }
+    );
+
+    if (theResolvedCase.outcome !== 'exactMatch') {
+      // If it is not an exact match the UI will ignore the `saved_object` data and redirect to the `alias_target_id`.
+      // We can return and omit the rest of the parameters since they will be requested again using the exactMatch ID.
+      return CaseResolveResponseRt.encode({
+        ...theResolvedCase,
+        saved_object: flattenCaseSavedObject({
+          savedObject: theResolvedCase.saved_object,
+        }),
+      });
+    }
+
+    let subCaseIds: string[] = [];
+    if (ENABLE_CASE_CONNECTOR) {
+      const subCasesForCaseId = await caseService.findSubCasesByCaseId({
+        unsecuredSavedObjectsClient,
+        ids: [id],
+      });
+      subCaseIds = subCasesForCaseId.saved_objects.map((so) => so.id);
+    }
+
+    await authorization.ensureAuthorized({
```

Comment:
hmm I think we should move the `ensureAuthorized` immediately after the `const theResolvedCase: SavedObjectsResolveResponse<CaseAttributes> = await caseService.getResolveCase(` but before any return statements.

Although if the outcome is a conflict or alias match I wonder if it's possible that the final case we're interested in could have a totally different `owner` 🤔 

I think if an unprivileged user requests a case with an old ID or a potential conflict, we should return an unauthorized error.

Thoughts @cnasikas ?

Reply:
Yes, makes complete sense to me. 
Regarding the owner, I don't think it can be different, but even if it can be different, I think we'll have to check the owner of the saved_object returned by the resolve function anyway.

Reply:
Yes, I totally agree. Every authorization check **MUST** be made before every other operation.